test/model/tools/model_test/main.cpp

```diff
@@ -0,0 +1,379 @@
+    fprintf(stderr, "  -?         show this message\n");
+}
+
+/*
+ * main --
+ *     The main entry point for the test.
+ */
+int
+main(int argc, char *argv[])
+{
+    model::kv_workload_generator_spec spec;
+    std::pair<uint64_t, uint64_t> p;
+    std::string home;
+    const char *progname;
+    bool preserve, print_only;
+    int ch;
+    uint64_t base_seed, min_iterations, min_runtime_s;
+
+    base_seed = (uint64_t)time(NULL);
+    home = "WT_TEST";
+    min_iterations = 1;
+    min_runtime_s = 0;
+    preserve = false;
+    print_only = false;
+    progname = argv[0];
+
+    std::string conn_config = ENV_CONFIG_BASE;
+    std::string table_config = TABLE_CONFIG_BASE;
+
+    /*
+     * Parse the command-line arguments.
+     */
+    try {
+        __wt_optwt = 1;
+        while ((ch = __wt_getopt(progname, argc, argv, "C:G:h:I:i:l:M:npS:T:t:?")) != EOF)
+            switch (ch) {
+            case 'C':
+                conn_config += ",";
+                conn_config += __wt_optarg;
+                break;
+            case 'G':
+                update_spec(spec, conn_config, table_config, __wt_optarg);
+                break;
+            case 'h':
+                home = __wt_optarg;
+                break;
+            case 'I':
+                min_iterations = parse_uint64(__wt_optarg);
+                break;
+            case 'i':
+                load_spec(spec, conn_config, table_config, __wt_optarg);
+                break;
+            case 'l':
+                p = parse_uint64_range(__wt_optarg);
+                spec.min_sequences = p.first;
+                spec.max_sequences = p.second;
+                if (p.first <= 0)
+                    throw std::runtime_error("Not enough transactions");
+                break;
+            case 'M':
+                p = parse_uint64_range(__wt_optarg);
+                spec.min_tables = p.first;
+                spec.max_tables = p.second;
+                if (p.first <= 0)
+                    throw std::runtime_error("Not enough tables");
+                break;
+            case 'n':
+                print_only = true;
+                break;
+            case 'p':
+                preserve = true;
+                break;
+            case 'S':
+                base_seed = parse_uint64(__wt_optarg);
+                break;
+            case 'T':
+                table_config += ",";
+                table_config += __wt_optarg;
+                break;
+            case 't':
+                min_runtime_s = parse_uint64(__wt_optarg);
+                break;
+            case '?':
+                usage(progname);
+                return EXIT_SUCCESS;
+            default:
+                usage(progname);
+                return EXIT_FAILURE;
+            }
+        argc -= __wt_optind;
+        if (argc != 0) {
+            usage(progname);
+            return EXIT_FAILURE;
+        }
+    } catch (std::exception &e) {
```

Comment:
A lot of catches in this function feel unnecessary, too (and also won't catch a `model_failure`, I think?)

Reply:
Why do you find most of these catches unnecessary? They all surround functions and blocks of code that can throw exceptions. I could, in theory, use one large try-catch block around bigger chunks of code, but I much prefer this approach, so that it is clearer from the error message what failed.

And yes, this would catch model failures, which throw instances of `model::verify_exception`.

Reply:
Sorry if that was unclear. My proposal was to not have a try/catch around them at all, and just let the failure do the default thing (print the message, then abort the program and possibly dump core).

Reply:
I guess I don't really see a utility in the program aborting and would rather have a clean error message, and also no Coverity issues (this is one of the things on which Coverity likes to pick up).

There are cases where having a core is useful for debugging, but I'm not sure how much this is the case here, as recovery is already long done by the time we would notice a verification error.